include/osquery/core.h

```diff
@@ -199,6 +199,12 @@ using WriteLock = boost::unique_lock<Mutex>;
 /// Helper alias for read locking a mutex.
 using ReadLock = boost::shared_lock<Mutex>;
 
+/// Helper alias for upgrade locking a mutex.
+using UpgradeLock = boost::upgrade_lock<Mutex>;
```

Comment:
I like upgrade ownership, but I'd like to keep it out of `core.h` if possible until you can find another use case. Placing this in core has dramatic negative effects on compile time. For now do you mind placing these into the `registry.cpp` implementation file?

Do you also mind moving the member function implementations from `registry.h` into `registry.cpp`?

This is a mild nit, but one that will have compound positive effects after time.

Reply:
Will do.